openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/local/VMLogger.java

```diff
@@ -0,0 +1,773 @@
+                }
+                
+                i++;                
+            }
+            
+        } catch (UnsupportedOperationException uoe) {
+            Message.logOut("One of the operations you tried is not supported");
+            uoe.printStackTrace();
+        }
+    }
+
+    public void disableOptionalFunctionality() {
+        try {
+            // ClassLoadingMXBean operations
+            mbs.setAttribute(srvClassName, new Attribute("Verbose", new Boolean(false)));
+
+            // MemoryMXBean operations
+            mbs.setAttribute(srvMemName, new Attribute("Verbose", new Boolean(false)));
+
+            // MemoryPoolMXBean operations
+            for (Object srvMemPoolObj: srvMemPoolNames) {
+                ObjectName srvMemPoolName = (ObjectName) srvMemPoolObj;
+
+                mbs.invoke(srvMemPoolName, "resetPeakUsage", new Object[] { }, new String[] { });
+
+                if (((Boolean)(mbs.getAttribute(srvMemPoolName, "CollectionUsageThresholdSupported"))).booleanValue()) {
+                    mbs.setAttribute(srvMemPoolName, new Attribute("CollectionUsageThreshold", new Long(0)));            }
+
+                if (((Boolean)(mbs.getAttribute(srvMemPoolName, "UsageThresholdSupported"))).booleanValue()) {
+                    mbs.setAttribute(srvMemPoolName, new Attribute("UsageThreshold", new Long(0)));
+                }            
+            }
+
+            // ThreadMXBean operations
+            mbs.invoke(srvThrdName, "resetPeakThreadCount", new Object[] { }, new String[] { });
+
+            if (((Boolean)(mbs.getAttribute(srvThrdName, "ThreadContentionMonitoringSupported"))).booleanValue()) {
+                mbs.setAttribute(srvThrdName, new Attribute("ThreadContentionMonitoringEnabled", new Boolean(false))); 
+            }
+
+
+            if (((Boolean)(mbs.getAttribute(srvThrdName, "ThreadCpuTimeSupported"))).booleanValue()) {
+                mbs.setAttribute(srvThrdName, new Attribute("ThreadCpuTimeEnabled", new Boolean(false)));           
+            }
+
+            // LoggingMXBean operations
+            String[] loggers = (String[]) mbs.getAttribute(srvLogName, "LoggerNames"); 
+            
+            String[] levels = {"FINEST", "FINER", "FINE", "CONFIG", "INFO", "WARNING", "SEVERE"};
+            int i = 0;
+            
+            for( String logger : loggers) {               	
+                if (i > 6) {
+                	i = i - 7;
+                }
+                
+                // LoggingMXBean operations
+                logBean.setLoggerLevel(logger, levels[i]);
+                mbs.invoke(srvLogName, "setLoggerLevel", new Object[] {logger, levels[i]}, new String[] {"java.lang.String", "java.lang.String"});
+                
+                i++;                
+            }
+       } catch (UnsupportedOperationException uoe) {
+            Message.logOut("One of the operations you tried is not supported");
```

Comment:
General comment for the logged messages in this file, that many of them do not seem very useful.  Do the stacktraces or some other info get automatically logged?

AttributeNotFoundException log message "Attribute does not exist on the MBean" does not really add value.  Based on how some of these tests were written, where try catch blocks around large pieces of code, that it may not be possible how it is structure to include more info (like what attribute, as there are 6 setAttribute calls in this block), but if logging is supposed to help debugging, we may have to revisit this.

Reply:
Agreed. The reporting and error logging in this test is rather messy. They generate "CSV" data files where some info is logged. The rest go into standard err and out. There are also some tests (e.g. VMLogger, where a log file is used to log detailed results of what the invoked Bean API's return. The output handling is not standardized in general. However, it is quite a lot of work to be honest to standardize the reporting mechanism of these tests. So, may be we could cover it in a separate enhancement work item?